desktop-widgets/downloadfromdivecomputer.cpp

```diff
@@ -149,6 +149,7 @@ int DownloadFromDCWidget::deviceIndex(QString deviceText)
 void DownloadFromDCWidget::DC##num##Clicked() \
 { \
 	ui.vendor->setCurrentIndex(ui.vendor->findText(qPrefDiveComputer::vendor##num())); \
+	ui.device->setCurrentText(qPrefDiveComputer::device##num()); \
```

Comment:
This seems to be doing the same as what we are already doing in https://github.com/subsurface/subsurface/pull/4120/files#diff-a0bde534c60cabf020b85f0079e0a8d567f5465514e8a59629b3a50a228d17f9R160 - but with slightly different connotations, like not overriding the selection for serial devices (which probably makes sense), and not doing it for MacOS where it does not seem to be working.

The core of the problem seems to be that https://github.com/subsurface/subsurface/pull/4120/files#diff-a0bde534c60cabf020b85f0079e0a8d567f5465514e8a59629b3a50a228d17f9L156 is triggering `enableBluetoothMode()`, which is springing a bluetooth selection dialogue - so I think it will be better if we just move this line down below `ui.device->setCurrentIndex()`.

Reply:
Thanks very much for the help and for looking at this.   I'm not sure why it should be different for serial vs Bluetooth devices, but I must be missing something.    In my test case, I have an Oceanic atom using /dev/tty.usbserial-20030001 and a G2 using a Bluetooth address.   With the changes above, I seem to be able to switch back and forth between the two without a Bluetooth scan or losing the serial mount point.    I'm doing on this on a sadly old Macbook running Catalina (10.15.7).    Won't your suggestion of moving this line down below ui.device-setCurrentIndex() be skipped on MacOs?

Reply:
The thinking behind not using persisted serial port data in #3840 is that on today's computers practically all serial devices are connected through USB, and on most operating systems have dynamic identifiers (based on the order they were detected in). So in most cases the serial port identifier may or may not have changed. Not sure how this is for MacOS, but the device identifier that you supplied looks a lot like it is dependent on the USB bus ID, which will change if a different port is used.
Re not using remembered bluetooth ids on MacOS, this is based on https://github.com/subsurface/subsurface/pull/2158#issuecomment-508933672, so you'll have to work out with @dirkhh if this is still relevant.

Reply:
@dirkhh: Do you have any insights about https://github.com/subsurface/subsurface/pull/2158#issuecomment-508933672 still being valid, or should this exclusion be dropped?

Reply:
From comments that others have made, and from a quick experiment, it seems like Apple may have fixed that.

Reply:
Sounds good @dirkhh, thanks for checking.
@notrege we still have the problem that this line essentially doubles up what https://github.com/subsurface/subsurface/pull/4120/files#diff-a0bde534c60cabf020b85f0079e0a8d567f5465514e8a59629b3a50a228d17f9R160 already does, and that it overwrites the serial port if it has already been selected, even if this serial port is not currently available.
So I'd probably remove this line, and instead drop the `!isMacOs` (and the comment about it) in the code below.